src/Common/src/Interop/Windows/Kernel32/Interop.QueryPerformanceCounter.cs

```diff
@@ -2,14 +2,21 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Runtime.InteropServices;
 
 internal partial class Interop
 {
     internal partial class Kernel32
     {
-        [DllImport(Libraries.Kernel32)]
-        internal static extern bool QueryPerformanceCounter(out long value);
+        // The actual native signature is:
+        //      BOOL WINAPI QueryPerformanceCounter(
+        //          _Out_ LARGE_INTEGER* lpPerformanceCount
+        //      );
+        //
+        // We take a long* (rather than a out long) to avoid the pinning overhead.
```

Comment:
Why is there pinning overhead for a `ref`/`out` parameter of blittable value types? What about a `in` parameter of `readonly struct`?

If so, why isn't the overhead optimized since they are equivalent?

Reply:
Because the value type could be a field of a reference type and therefore require pinning.

It is cheaper to just always pin then to check if something needs to be pinned, and so not using ref/out/in avoids it implicitly happening and makes it explicit by the caller (and therefore only happening where requires).

Reply:
If the value type is a field of a reference type then pinning is required anyway.

Otherwise will it get pinned as a by-reference parameter?